checks/cs50/2017/x/recover/checks.py

```diff
@@ -0,0 +1,68 @@
+import os
+import sys
+import subprocess
+
+sys.path.append(os.getcwd())
+from check50 import *
+
+class Recover(TestCase):
+
+    @check()
+    def exists(self):
+        """recover.c exists."""
+        super().exists("recover.c")
+
+    @check("exists")
+    def compiles(self):
+        """recover.c compiles."""
+        self.spawn("clang -o recover recover.c -lcs50 -lm").exit(0)
+
+    @check("compiles")
+    def test_noimage(self):
+        """handles lack of forensic image"""
+        self.spawn("./recover").exit(1)
+
+    @check("compiles")
+    def first_image(self):
+        """recovers 000.jpg correctly"""
+        self.include("card.raw")
+        subprocess.call("bash", shell=True)
+        self.spawn("./recover card.raw").exit(0)
+        hash000 = 'c3ceda9dde5cf2212e96d6f61e2e849ac60e2c3cbdf0184f6e0c3b6ea5e69d31'
+        if self.hash("000.jpg") != hash000:
+            raise Error("recovered image does not match")
+
+    @check("compiles")
+    def middle_images(self):
+        """recovers middle images correctly"""
+        self.include("card.raw")
```

Comment:
@crossroads1112 should the semantics here perhaps be `self.copy` or `self.import` instead? Not sure, just had to think twice about what `include` meant here.